rs/bitcoin/adapter/src/header_cache.rs

```diff
@@ -709,6 +712,7 @@ pub(crate) mod test {
                     next_headers.insert(header.block_hash(), header);
                 }
             }
+            assert_eq!(cache.metrics.in_memory_elements.get(), 7);
```

Comment:
nit: let's also check that the on disk elements did not change

```suggestion
            assert_eq!(cache.metrics.in_memory_elements.get(), 7);
            assert_eq!(cache.metrics.on_disk_elements.get(), 1);
```